### PR TITLE
Qt6 another try

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by nicogodet** [comment]
https://github.com/3nids/QGIS/blob/qt6-another-try/.github/workflows/run-tests.yml#L61-L70

File: .docker/qgis3-qt6-build-deps.dockerfile [hunk 1]
```diff
@@ -45,16 +45,18 @@ RUN dnf -y install \
...
-  && wget https://github.com/KDE/qca/archive/refs/tags/v2.3.3.zip \
-  && unzip v2.3.3.zip \
-  && mkdir build-qt6 \
-  && cd build-qt6 \
+  && wget https://github.com/KDE/qca/archive/refs/heads/master.zip \
+  && unzip master.zip \
+  && rm master.zip \
+  && mkdir build \
+  && cd build \
   && cmake -DCMAKE_INSTALL_PREFIX=/usr -DQT6=ON -GNinja ../qca-2.3.3 \
```
Comment (nicogodet) [inline .docker/qgis3-qt6-build-deps.dockerfile:53]:
```suggestion
  && cmake -DCMAKE_INSTALL_PREFIX=/usr -DQT6=ON -GNinja ../qca-master \
```

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -58,16 +58,16 @@ jobs:
...
-          #   qt-version: 6
-          #   run-tests: false
-          #   docker-tag-suffix: ''
-          #   with-qt6: ON
-          #   with-qt5: OFF
-          #   with-3d: OFF
-          #   with-quick: OFF
-          #   patch-qt-3d: false
-          #   LD_PRELOAD: ''
+          - distro-version: 'rawhide'
```
Comment (nirvn) [inline .github/workflows/run-tests.yml:61]: @3nids , will this need to be updated to 34 too?

**Review by nirvn — APPROVED** [review]

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (3nids) [reply .github/workflows/run-tests.yml:61]: not strictly needed, right now (as it's not yet a parameter in the Dockerfile), let's wait for the PR to be green

**Comment by nicogodet** [comment]
I also notice this
https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L109
`image_name: qgis3-qt${{ matrix.qt-version }}-build-deps`

https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L118
`docker tag ${DOCKER_IMAGE} qgis3-qt${{ matrix.qt-version }}-build-deps`

https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L168
`qgis3-qt${{ matrix.qt-version }}-build-deps \`

**Comment by 3nids** [comment]
> I also notice this
> https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L109
> 
> 
> `image_name: qgis3-qt${{ matrix.qt-version }}-build-deps`
> https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L118
> 
> 
> `docker tag ${DOCKER_IMAGE} qgis3-qt${{ matrix.qt-version }}-build-deps`
> https://github.com/qgis/QGIS/blob/1dd295e587518394df8842df4d02612e4302ed8d/.github/workflows/run-tests.yml#L168
> 
> 
> `qgis3-qt${{ matrix.qt-version }}-build-deps \`

well spotted, but that's meant to be. We currently push to Docker hub only one image.

**Comment by 3nids** [comment]
depending on #44728 for now

**Comment by nyalldawson** [comment]
We should set BUILD_TESTS to off for qca -- that'll save some time

**Comment by nyalldawson** [comment]
Same with examples and tests for.qwt

**Comment by 3nids** [comment]
build succeeded with Qt6 💪 
https://github.com/qgis/QGIS/pull/44574/checks?check_run_id=3349710598

**Comment by nirvn** [comment]
YAY!

**Comment by nirvn** [comment]
@3nids , what prevents us from building analysis on Qt6?

**Comment by 3nids** [comment]
@nirvn not sure anymore, let's move this to a new PR

**Pull request merged**